client/network/src/discovery.rs

```diff
@@ -159,6 +151,7 @@ impl DiscoveryConfig {
 
 		let mut config = KademliaConfig::default();
 		config.set_protocol_name(proto_name);
+		config.set_kbucket_inserts(KademliaBucketInserts::Manual);
```

Comment:
I would suggest a small comment here, to explain what this does without having to look at the documentation of `libp2p-kad`.

```suggestion
		// Configures Kademlia to not automatically insert in the DHT nodes we connect to.
		config.set_kbucket_inserts(KademliaBucketInserts::Manual);
```